test/common/crudtester.go

```diff
@@ -714,27 +715,133 @@ func (c *FederatedTypeCrudTester) versionForCluster(version *fedv1a1.PropagatedV
+
+	client := c.resourceClient(apiResource)
+	var remoteStatusObj interface{}
+	// The default is normally 30 seconds
+	waitTimeout := 6 * wait.ForeverTestTimeout
+	err := wait.PollImmediate(c.waitInterval, waitTimeout, func() (bool, error) {
+		fedObj, err := client.Resources(qualifiedName.Namespace).Get(context.Background(), qualifiedName.Name, metav1.GetOptions{})
+		if err != nil {
+			c.tl.Errorf("An unexpected error occurred while polling for remote status: %v", err)
+			return false, nil
+		}
+
+		if err == nil {
+			resource := &status.GenericFederatedResource{}
+			err := util.UnstructuredToInterface(fedObj, resource)
+			if err != nil {
+				return false, err
+			}
+			if resource.Status != nil {
+				for _, cluster := range resource.Status.Clusters {
+					c.tl.Logf("Current status of resource for cluster '%s' with value: %v", cluster.Name, resource.Status)
+					if cluster.Name == clusterName && cluster.Status == status.ClusterPropagationOK {
+						c.tl.Logf("resource remote status for cluster '%s': %v", cluster.Name, cluster.RemoteStatus)
+						if cluster.RemoteStatus != nil {
+							remoteStatusObj = cluster.RemoteStatus
+							return true, nil
+						}
+					}
+				}
+			}
+			return false, nil
+		}
+		return false, err
+	})
+
+	if err != nil {
+		c.tl.Fatalf("Timed out waiting for the remote status: %q", qualifiedName)
+		return nil, err
+	}
+
+	return remoteStatusObj, nil
+}
+
 func (c *FederatedTypeCrudTester) CheckStatusCreated(qualifiedName util.QualifiedName) {
 	if !c.typeConfig.GetStatusEnabled() {
 		return
 	}
 
 	statusAPIResource := c.typeConfig.GetStatusType()
-	statusKind := statusAPIResource.Kind
+	// TODO (hectorj2f): To be deprecated.
+	// I added a check to support current FederatedServiceStatus and new remote resource
+	// status. The new mechanism reuses the statusCollection flag but it does not define
+	// a StatusType type.
+	if statusAPIResource != nil {
+		statusKind := statusAPIResource.Kind
 
-	c.tl.Logf("Checking creation of %s %q", statusKind, qualifiedName)
+		c.tl.Logf("Checking creation of %s %q", statusKind, qualifiedName)
 
-	client := c.resourceClient(*statusAPIResource)
-	err := wait.PollImmediate(c.waitInterval, wait.ForeverTestTimeout, func() (bool, error) {
-		_, err := client.Resources(qualifiedName.Namespace).Get(context.Background(), qualifiedName.Name, metav1.GetOptions{})
-		if err != nil && !apierrors.IsNotFound(err) {
-			c.tl.Errorf("An unexpected error occurred while polling for desired status: %v", err)
-		}
-		return (err == nil), nil
-	})
+		client := c.resourceClient(*statusAPIResource)
+		err := wait.PollImmediate(c.waitInterval, wait.ForeverTestTimeout, func() (bool, error) {
+			_, err := client.Resources(qualifiedName.Namespace).Get(context.Background(), qualifiedName.Name, metav1.GetOptions{})
+			if err != nil && !apierrors.IsNotFound(err) {
+				c.tl.Errorf("An unexpected error occurred while polling for desired status: %v", err)
+			}
+			return (err == nil), nil
```

Comment:
```suggestion
			return err == nil, nil
```
parens are not needed